qemu/tests/virtio_serial_throttling.py

```diff
@@ -0,0 +1,57 @@
+from virttest import error_context, env_process
+from virttest.utils_virtio_port import VirtioPortTest
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Induce throttling by opening chardev but not reading data from it:
+    1) Start guest with virtio-serial with unix option
+    2) On the host, open the chardev but don't read from it
+    3) In the guest, write to the virtio-serial port.
+    4) repeat step 1 to 3 with tcp option.
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    
+    def send_data_from_guest_to_host(serial_name):
+        port.open()
+        session.cmd_output('dd if=/dev/zero of=/dev/%s bs=1024 count=1' % serial_name)
+
+    def send_data_from_host_to_guest():
+        port.open()
+        port.sock.send(b'Hello world \n')
+        guest_worker.cmd("virt.open('%s')" % port.name)
+
+    def clean_env():
+        guest_worker.cleanup()
+        port.close()
+
+
+    for j in range(2):
+        virtio_test = VirtioPortTest(test, env, params)
+        (vm, guest_worker, port) = virtio_test.get_vm_with_single_port()
+        vm.verify_alive()
+        serial_id = params.objects('serials')[-1]
+        session = vm.wait_for_login()
+        if j == 0:
+            send_data_from_guest_to_host(port.name)
+        else:
+            send_data_from_host_to_guest()
+        clean_env()
+        params['chardev_backend_%s' % serial_id] = 'tcp_socket'
```

Comment:
did not destroy vm, but boot it again.
And, in this way:
1. unix_socket, guest-> host
2. tcp_scoket, guest->host
3. unix_socket, host -> guest
4. tcp_scoket, host -> guest

It need destroy vm for 3 times during the test.
we can use test one backend with guest-> host, and host-> guest, then next backend.